helios-core/src/main/kotlin/helios/core/helios.kt

```diff
@@ -98,10 +99,8 @@ sealed class Json {
     (this as? JsNull)?.some() ?: none()
 
   fun merge(that: Json): Json =
-    Option.applicative().map(asJsObject(), that.asJsObject()) { (lhs, rhs) ->
-      lhs.toList().fold(rhs) { acc, (key, value) ->
-        rhs[key].fold({ acc.add(key, value) }, { r -> acc.add(key, value.merge(r)) })
-      }
+    Option.applicative().map(asJsObject(), that.asJsObject()) {
+      JsObject(it.a.value + it.b.value)
```

Comment:
This changes the behavior of the previous `merge`, this indicates a different issue however.
`merge` should be parameterized with a strategy of combining elements, `Monoid`.

`fun merge(that: Json, M: Monoid<Json>): Json`.

If you have any questions feel free to poke here or on Slack ;)